smartseq2_single_sample/run_multi_samples/scripts/calculate_cost_workflow.py

```diff
@@ -0,0 +1,253 @@
+import argparse
+import json
+import math
+import urllib.request, urllib.error, urllib.parse
+import dateutil.parser
+from io import StringIO
+import gzip
+import requests
+
+GCE_MACHINE_TYPES_URL = "http://cloudpricingcalculator.appspot.com/static/data/pricelist.json"
+TOTAL_WORKFLOW_COST = 0
+
+# load the US `pricing for both persistent disk and compute engine
+def get_gce_pricing():
+    response = urllib.request.urlopen(GCE_MACHINE_TYPES_URL)
+    data = response.read()
+
+    if response.info().get('Content-Encoding') == 'gzip':
+        buf = StringIO(data)
+        f = gzip.GzipFile(fileobj=buf)
+        data = f.read()
+
+    pricing = json.loads(data)
+
+    data = {}
+    for k, v in pricing.items():
+        if k == "gcp_price_list":
+            for k2, v2 in v.items():
+                if k2.startswith("CP-COMPUTEENGINE-VMIMAGE"):
+                    data[k2.replace("CP-COMPUTEENGINE-VMIMAGE-", "").lower()] = v2['us']
+                if k2.startswith("CP-COMPUTEENGINE-STORAGE-PD"):
+                    data[k2] = v2['us']
+                if k2.startswith("CP-COMPUTEENGINE-STORAGE-PD-CAPACITY"):
+                    data[k2] = v2['us']
+
+    return data
+
+
+def extract_machine_type(t):
+    base_machine_type = "unknown"
+
+    if 'jes' in t and 'machineType' in t['jes']:
+        full_machine = t['jes']['machineType']
+        (zone, base_machine_type) = full_machine.split("/")
+
+    return base_machine_type
+
+
+def get_disk_info(metadata):
+    if "runtimeAttributes" in metadata and "disks" in metadata['runtimeAttributes']:
+        bootDiskSizeGb = 0.0
```

Comment:
It would be more consistent to use underscores in the variable name here, `boot_disk_size_gb`